packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -99,6 +130,77 @@ export const getKey = (node: t.Expression) => {
   throw new Error();
 };
 
+/**
+ * Will recursively traverse a path and its identifiers to find all bindings.
+ * If any of those bindings are mutable `true` will be returned.
+ * Else `false`.
+ *
+ * @param path
+ */
+export const doesPathReferenceAnyMutableIdentifiers = (path: NodePath<any>): boolean => {
+  if (path.isIdentifier()) {
+    const binding = path.scope.getBinding(path.node.name);
+    if (!binding || !t.isVariableDeclarator(binding.path.node) || binding.kind !== 'const') {
```

Comment:
nit: `binding.kind !== 'const'` can be `!binding.constant`

Reply:
interestingly tests start failing when making this change - maybe they're for two different things?

Reply:
😱 It was given here so I thought https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#bindings

Reply:
ohh got it. It refers to variable which is not assigned again. Even if it is `let`

Reply:
Can we make good use of 

```typescript
 constant: false,
constantViolations: [path]
```

somehow?

Reply:
hmm let me look at what `constantViolations` pumps out - wonder what it's for?

Reply:
I think this one:

```typescript
function scopeOne() {
  var ref1 = "This is a constant binding";

  becauseNothingEverChangesTheValueOf(ref1);

  function scopeTwo() {
    var ref2 = "This is *not* a constant binding";
    ref2 = "Because this changes the value";
  }
}
```

Reply:
OH! so we can safely use `binding.constant` - it basically returns `true` if it isnt mutated - and false otherwise, maybe?

sounds good! we should use it! let me test

Reply:
yups. I think its for that purpose only.

Reply:
yeah thats exactly what happens. awesome